src/core/HY_Features.cpp

```diff
@@ -3,59 +3,67 @@
-        feat_id = network.get_id(feat_idx);//feature->get_id();
-        feat_type = feat_id.substr(0, 3);
-
-        destinations  = network.get_destination_ids(feat_id);
-        if(feat_type == "cat")
-        {
-          //Find and prepare formulation
-          auto formulation = formulations->get_formulation(feat_id);
-          formulation->set_output_stream(formulations->get_output_root() + feat_id + ".csv");
-          // TODO: add command line or config option to have this be omitted
-          //FIXME why isn't default param working here??? get_output_header_line() fails.
-          formulation->write_output("Time Step,""Time,"+formulation->get_output_header_line(",")+"\n");
-          //Find upstream nexus ids
-          origins = network.get_origination_ids(feat_id);
-          //Create the HY_Catchment with the formulation realization
-          std::shared_ptr<HY_Catchment> c = std::make_shared<HY_Catchment>(
-              HY_Catchment(feat_id, origins, destinations, formulation)
-            );
-
-          _catchments.emplace(feat_id, c);
-        }
-        else if(feat_type == "nex" || feat_type == "tnx")
-        {
-            _nexuses.emplace(feat_id, std::make_unique<HY_PointHydroNexus>(
-                                          HY_PointHydroNexus(feat_id, destinations) ));
-        }
-        else
-        {
-          std::cerr<<"HY_Features::HY_Features unknown feature identifier type "<<feat_type<<" for feature id."<<feat_id
-                   <<" Skipping feature"<<std::endl;
-        }
-      }
-
+  init();
 }
 
-HY_Features::HY_Features( geojson::GeoJSON catchments, std::string* link_key, std::shared_ptr<Formulation_Manager> formulations):
-    HY_Features(network::Network(catchments, link_key), formulations){
+void HY_Features::init()
+{
+  std::string feat_id;
+  std::string feat_type;
+  std::vector<std::string> origins, destinations;
+
+  for(const auto& feat_idx : network)
+  {
+    feat_id = network.get_id(feat_idx);//feature->get_id();
+    feat_type = feat_id.substr(0, 3);
+
+    destinations  = network.get_destination_ids(feat_id);
+    if(feat_type == "cat" || feat_type == "agg")
+    {
+      //Find and prepare formulation
+      auto formulation = formulations->get_formulation(feat_id);
+      formulation->set_output_stream(formulations->get_output_root() + feat_id + ".csv"); // MERGE bd63e5b
+      // TODO: add command line or config option to have this be omitted
+      //FIXME why isn't default param working here??? get_output_header_line() fails.
+      formulation->write_output("Time Step,""Time,"+formulation->get_output_header_line(",")+"\n");
+      //Find upstream nexus ids
+      origins = network.get_origination_ids(feat_id);
+
+      // get the catchment level from the hydro fabric
+      const auto& cat_json_node = fabric->get_feature(feat_id);
+      long lv = cat_json_node->has_key("level") ? cat_json_node->get_property("level").as_natural_number() : 0;
```

Comment:
I was going through and unifying "level" to "layer" for consistency... but this appears it would actually have a functional impact, because the provided /data/catchment_data_multilayer.geojson file actually _does_ have `"layer"` keys instead of `"level"`... of course the test code was previously looking for a different example file which was not committed, so I don't know what was in that one.  @donaldwj can you look at this please? I will test some with this but am not as confident I'll figure out the impact.

Reply:
After changing this, it _also_ passes the included test with the example file that has `"layer"`... for better and worse.